src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -146,7 +169,7 @@ public InfluxDBImpl(final String url, final String username, final String passwo
       break;
     }
 
-    this.retrofit = new Retrofit.Builder().baseUrl(url).client(
+    this.retrofit = retrofitBuilder.baseUrl(url).client(
```

Comment:
Just a heads up: you may have here `ConcurrentModificationException` as you are adding a `converterFactory` to a shared and non thread-safe `retrofitBuilder`. Worst case scenatio if you don't trigger a CME, you will end up having multiple `converterFactory` [added to the internal list in retrofitBuilder](https://github.com/square/retrofit/blob/master/retrofit/src/main/java/retrofit2/Retrofit.java#L515).

Reply:
Yes, you are right. It could cause same errors as in #445, so I will use same approach to avoid potential problems:

```java
Retrofit.Builder clonedRetrofitBuilder = retrofitBuilder.baseUrl(url).build().newBuilder();
this.retrofit = clonedRetrofitBuilder.client(clonedOkHttpBuilder.build())
            .addConverterFactory(converterFactory).build();
```